client/src/lib.rs

```diff
@@ -115,9 +116,23 @@ async fn execute_task<'a, T, E: From<ClientError>>(
     task.init(Some(alloc))?;
     let mut cont = TaskResult::Continue;
     while cont == TaskResult::Continue {
-        while wait_preemptive(client, timeout).await? {
-            tokio::time::sleep(Duration::from_millis(1000)).await
+        let mut wait = true;
+        while wait {
+            let preemtive_state = wait_preemptive(client, timeout).await?;
+            match preemtive_state {
+                PreemptionResponse::Wait => tokio::time::sleep(Duration::from_millis(1000)).await,
+                PreemptionResponse::Execute => {
+                    //final check client-side execution?
+                    wait = false;
+                }
+                PreemptionResponse::Abort => {
+                    warn!("Client {} aborted", client.token.process_id());
+                    release_preemptive(client).await?;
+                    return task.end(Some(alloc));
```

Comment:
Should we return an error here? returning an error on abort is better, we can add a new error variant so clients would know what to do.